### ISSUE TITLE
Forgot password function does not work with some account

### ISSUE THREAD

**Issue by jcwerck** [salesagility/SuiteCRM#3683]
#### Issue
When one of the users use the forgot password function, it doesn't work.
#### Expected Behavior
Sending an email with the link to reset the password

#### Actual Behavior
Error message : "Provide both a User Name and an Email Addres"

#### Possible Fix
?

#### Steps to Reproduce
It works for others users. I don't know why it does not work with this user.

#### Your Environment
* SuiteCRM Version used: 7.9.0
* Browser name and version : Chrome Version 58.0.3029.110 (64-bit)
* Environment name and version : MySQL 5.1.73, PHP 5.3.3, Apache 2.2.15
* Operating System and version : CentOS 6.6

I have compared the queries on the database for the user for which the forgot password can not be retrieved and for on for which it works.

**When it works :**
```
SELECT users.* FROM users  WHERE user_name = 'username' AND deleted=0 LIMIT 0,1
SELECT id, first_name, last_name, user_name FROM users WHERE 1=1 ORDER BY  user_name ASC
SELECT u1.first_name, u1.last_name from users  u1, users  u2 where u1.id = u2.reports_to_id AND u2.id = '2b7bb4e4-6e82-a6b6-b68f-58c680803efd' and u1.deleted=0
SELECT users.* FROM users  WHERE users.id = '2b7bb4e4-6e82-a6b6-b68f-58c680803efd' AND users.deleted=0 LIMIT 0,1
SELECT u1.first_name, u1.last_name from users  u1, users  u2 where u1.id = u2.reports_to_id AND u2.id = '2b7bb4e4-6e82-a6b6-b68f-58c680803efd' and u1.deleted=0
SELECT ea.email_address, ea.email_address_caps, ea.invalid_email, ea.opt_out, ea.date_created, ea.date_modified,
                ear.id, ear.email_address_id, ear.bean_id, ear.bean_module, ear.primary_address, ear.reply_to_address, ear.deleted
                FROM email_addresses ea LEFT JOIN email_addr_bean_rel ear ON ea.id = ear.email_address_id
                WHERE ear.bean_module = 'Users'
                AND ear.bean_id = '2b7bb4e4-6e82-a6b6-b68f-58c680803efd'
                AND ear.deleted = 0
                ORDER BY ear.reply_to_address, ear.primary_address DESC
SELECT contents FROM user_preferences WHERE assigned_user_id='2b7bb4e4-6e82-a6b6-b68f-58c680803efd' AND category = 'global' AND deleted = 0
SELECT ea.email_address FROM email_addresses ea
                LEFT JOIN email_addr_bean_rel ear ON ea.id = ear.email_address_id
                WHERE ear.bean_module = 'Users'
                AND ear.bean_id = '2b7bb4e4-6e82-a6b6-b68f-58c680803efd'
                AND ear.deleted = 0
                AND ea.invalid_email = 0
                ORDER BY ear.primary_address DESC LIMIT 0,1
INSERT INTO users_password_link (id, username, date_generated) VALUES('....
```

**When it fails :** 
```
SELECT users.* FROM users  WHERE user_name = 'username2' AND deleted=0 LIMIT 0,1
SELECT id, first_name, last_name, user_name FROM users WHERE 1=1 ORDER BY  user_name ASC
SELECT u1.first_name, u1.last_name from users  u1, users  u2 where u1.id = u2.reports_to_id AND u2.id = '3791e868-aafb-1ad6-69c3-58cbab80b566' and u1.deleted=0`
SELECT users.* FROM users  WHERE users.id = '3791e868-aafb-1ad6-69c3-58cbab80b566' AND users.deleted=0 LIMIT 0,1
SELECT u1.first_name, u1.last_name from users  u1, users  u2 where u1.id = u2.reports_to_id AND u2.id = '3791e868-aafb-1ad6-69c3-58cbab80b566' and u1.deleted=0
SELECT ea.email_address, ea.email_address_caps, ea.invalid_email, ea.opt_out, ea.date_created, ea.date_modified,
                ear.id, ear.email_address_id, ear.bean_id, ear.bean_module, ear.primary_address, ear.reply_to_address, ear.deleted
                FROM email_addresses ea LEFT JOIN email_addr_bean_rel ear ON ea.id = ear.email_address_id
                WHERE ear.bean_module = 'Users'
                AND ear.bean_id = '3791e868-aafb-1ad6-69c3-58cbab80b566'
                AND ear.deleted = 0
                ORDER BY ear.reply_to_address, ear.primary_address DESC
SELECT contents FROM user_preferences WHERE assigned_user_id='3791e868-aafb-1ad6-69c3-58cbab80b566' AND category = 'global' AND deleted = 0
```

The last query return something. And the email retrieved in the database with the username is the same as the email entered in the "Forgot password" fields.

**Comment by Dillon-Brown** [comment]
Hi @jcwerck has this user been previously created and deleted with the same username? if so, I would try out the fix for #3440. Let me know how you get on.

**Comment by jcwerck** [comment]
Hi @Dillon-Brown. No, it is not the case. But I find that there were 2 records in the database for the email adress of the user. One is linked with one Contact and the User, the second is linked with one Contact.
I have deleted the corresponding rows for this two addresses in email_addresses and email_addr_bean_rel. After that, I recreated the email address for the user, and the Forgot password functionnality works (it takes a long time and finally I have a popup "Undefined" but the email is sent).
